### PR TITLE
Remove client error logs

### PR DESCRIPTION
**Pull request by VivekVinushanth**
### Proposed changes in this pull request
This PR removes logging client errors.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java [hunk 1]
```diff
@@ -153,7 +153,9 @@ private CarbonOAuthTokenRequest handleInvalidRequest(OAuthProblemException e)
                 log.debug("Error: " + e.getError() + ", description: " + e.getDescription());
             }
         } else {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java:155]: We can remove the else block and move the logging outside

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java [hunk 2]
```diff
@@ -1199,9 +1199,14 @@ public static String getSPTenantDomainFromClientId(String clientId) {
         try {
             OAuthAppDO oAuthAppDO = OAuth2Util.getAppInformationByClientId(clientId);
             return OAuth2Util.getTenantDomainOfOauthApp(oAuthAppDO);
-        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+        } catch (IdentityOAuth2Exception e) {
             log.error("Error while getting oauth app for client Id: " + clientId, e);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java:1203]: If we are doing a fallback here, shouldn't this be a WARN or a DEBUG log?

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java (hunk 1 above)
Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java:155]: The  logging has to be different as per the existing logic. Hence, else part is needed IMO.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java [hunk 3]
```diff
@@ -139,10 +139,15 @@ public Scope registerScope(Scope scope) throws IdentityOAuth2ScopeException {
...
-        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+        } catch (IdentityOAuth2Exception e) {
             log.error("Error while getting oauth app for client Id: " + clientId, e);
             throw Oauth2ScopeUtils.generateServerException(Oauth2ScopeConstants.ErrorMessages.
                     ERROR_CODE_FAILED_TO_GET_ALL_SCOPES, e);
+        } catch (InvalidOAuthClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error while getting oauth app for client Id: " + clientId, e);
+            }
+            return null;
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java:150]: don't we have IdentityOAuth2ScopeClientException to through from here, so the API layer will handle it properly

Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java:150]: This is a public method and it throws only a server exception. So, though we can throw a client exception we might need to modify method signature which am bit skeptical about.

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java (hunk 2 above)
Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java:1264]: We are separating client and server errors. **IdentityOAuth2Exception**  is considered to be server error and we also catch **InvalidOAuthClientException** for which we will have debug logs.

**Review by senthalan — APPROVED** [review]

**Pull request merged**